cobigen-plugins/cobigen-templateengines/cobigen-tempeng-agnostic/src/main/java/com/devonfw/cobigen/tempeng/agnostic/AgnosticTemplateEngine.java

```diff
@@ -0,0 +1,199 @@
+
+    this(null);
+  }
+
+  /**
+   * Constructor.
+   */
+  public AgnosticTemplateEngine(ClassLoader classLoader) {
+
+    super();
+    this.classLoader = classLoader;
+  }
+
+  @Override
+  public void setTemplateFolder(Path templateFolderPath) {
+
+    // pointless
+  }
+
+  @Override
+  public String getTemplateFileEnding() {
+
+    return null;
+  }
+
+  @Override
+  public void process(TextTemplate template, Map<String, Object> modelAsMap, Writer writer, String outputEncoding) {
+
+    try {
+      CobiGenModelDefault model = new CobiGenModelDefault(modelAsMap);
+      process(template, model, writer);
+    } catch (Throwable e) {
+      throw new CobiGenRuntimeException("An unkonwn error occurred while generating the template."
+          + template.getAbsoluteTemplatePath() + "(Agnostic)", e);
+    }
+  }
+
+  private void process(TextTemplate template, CobiGenModel model, Writer writer) {
+
+    String path = template.getRelativeTemplatePath();
+    path = model.resolve(path, '.', VariableSyntax.AGNOSTIC);
+    AbstractCobiGenOutput out = (AbstractCobiGenOutput) CobiGenOutputFactory.get().create(path);
+    model = new CobiGenModelDefault(model);
+    CobiGenVariableDefinitions.OUT.setValue(model, out);
+    if (out instanceof StreamingCobiGenOutput) {
+      out = null;
+    }
+    AbstractCobiGenOutput currentOut = out;
+    Path templatePath = template.getAbsoluteTemplatePath();
+    try (BufferedReader reader = Files.newBufferedReader(templatePath)) {
+      boolean todo = true;
+      while (todo) {
+        String line = reader.readLine();
+        if (line != null) {
+          line = processLine(line, model, writer, out);
+          if (line != null) {
+            if (currentOut == null) {
+              writer.write(line);
+              writer.write('\n');
+            } else {
+              AbstractCobiGenOutput newOut = currentOut.addLine(line);
+              if (newOut != currentOut) {
+                currentOut = newOut;
+                model = new CobiGenModelDefault(model);
+                CobiGenVariableDefinitions.OUT.setValue(model, currentOut);
+              }
+            }
+          }
+        } else {
+          todo = false;
+        }
+      }
+      if (out != null) {
+        out.write(writer);
+      }
+    } catch (IOException e) {
+      throw new IllegalStateException("I/O error while instantiation template.");
+    }
+  }
+
+  private String processLine(String line, CobiGenModel model, Writer writer, AbstractCobiGenOutput out)
+      throws IOException {
+
+    Matcher matcher = PATTERN_COBIGEN.matcher(line);
+    boolean cgFound = matcher.find();
+    if (cgFound) {
+      if (line.trim().startsWith("import")) {
+        return null;
+      }
+      StringBuilder sb = null;
+      do {
+        String cobiGenString = matcher.group();
+        String cobiGenType = matcher.group(1);
+        CobiGenGenerator generator;
+        if (cobiGenString.startsWith("@")) {
+          if (sb == null) {
```

Comment:
Lift is giving me really stupid false positives so I feel like alpha-tester of lift rather than getting any value out of it. After twenty years of checkstyle, PMD, findbugs/spotbugs, sonar(qube), etc. static code analysis for Java code-quality should not be an experimental rocket-science ;)